### DISCUSSION TITLE
How to use buildToolsRepository for help index?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#4242] [Q&A] [answered]
This night all our company builds failed in the help index mojo: Execution eclipse-help-index of goal org.eclipse.tycho.extras:tycho-document-bundle-plugin:4.0.8:build-help-index failed: Failed to load p2 repository from location https://download.eclipse.org/releases/2024-03/: Unable to read repository at https://download.eclipse.org/releases/2024-03

I assume that this was a temporary issue. However, I would like to use the buildToolsRepository configuration parameter and use a local mirror to avoid the same in the past. I tried with this config (creating a repository node, and referring to its id in the plugin config):
```
	<repositories>
		<repository>
			<id>tycho-eclipse-buildtools</id>
			<url>https://company/artifactory/eclipse/releases/2024-06/</url>
		</repository>
	</repositories>

					<plugin>
						<groupId>org.eclipse.tycho.extras</groupId>
						<artifactId>tycho-document-bundle-plugin</artifactId>
						<executions>
							<execution>
								<id>eclipse-help-index</id>
								<goals>
									<goal>build-help-index</goal>
								</goals>
							</execution>
						</executions>
						<configuration>
							<buildToolsRepository>tycho-eclipse-buildtools</buildToolsRepository>
						</configuration>
					</plugin>

```
But that gives me `Unable to parse configuration of mojo org.eclipse.tycho.extras:tycho-document-bundle-plugin:4.0.8:build-help-index for parameter buildToolsRepository: Cannot find default setter in class org.apache.maven.model.Repository`. Also using the URL directly didn't work. What's the expected way to configure such a build tools repository?

**Comment by laeubi** [comment, marked as answer]
```
<plugin>
	<groupId>org.eclipse.tycho.extras</groupId>
	<artifactId>tycho-document-bundle-plugin</artifactId>
	<executions>
		<execution>
			<id>eclipse-help-index</id>
			<goals>
				<goal>build-help-index</goal>
			</goals>
		</execution>
	</executions>
	<configuration>
		<buildToolsRepository>
			<id>tycho-eclipse-buildtools</id>
			<url>https://company/artifactory/eclipse/releases/2024-06/</url>
		</buildToolsRepository>
	</configuration>
</plugin>
```

By the way as far as I know at laest eclipse stopped generating a help index and let it be performed at runtime.
